src/main/java/ru/hogwarts/school/service/AvatarService.java

```diff
@@ -0,0 +1,73 @@
+package ru.hogwarts.school.service;
+
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import ru.hogwarts.school.model.Avatar;
+import ru.hogwarts.school.model.AvatarDTO;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repository.AvatarRepository;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static java.nio.file.StandardOpenOption.CREATE_NEW;
+
+@Service
+@Transactional
+public class AvatarService {
+    @Value("${path.to.avatars.folder}")
+    private String avatarsDir;
+    private final AvatarRepository avatarRepository;
+    private final StudentService studentService;
+    private final AvatarDTO avatarDTO;
+
+    public AvatarService(String avatarsDir, AvatarRepository avatarRepository, StudentService studentService, AvatarDTO avatarDTO) {
```

Comment:
Нужно перенести аннотацию `@Value` сюда
```suggestion
    public AvatarService(@Value("${path.to.avatars.folder}") String avatarsDir, AvatarRepository avatarRepository, StudentService studentService, AvatarDTO avatarDTO) {
```